File: Java/src/main/java/codecraft/ticketbooker/ConsoleApp.java
package codecraft.ticketbooker;

import static java.lang.System.*;
import java.util.Scanner;

public class ConsoleApp
{
    private static Tickets Tickets = new Tickets();
    public static void main(String[] args)
    {
        String consoleInput;
        Scanner scanner = new Scanner(System.in);
        do
        {
            consoleInput = scanner.nextLine();
            if (consoleInput.contains("print summary"))
            {
                out.println();
                out.println(Tickets.GetReport());
            }
            else if (consoleInput.contains("set ticket price"))
            {
                String[] enteredTextArray = consoleInput.split(" ");
                Tickets.TicketPrice = Integer.parseInt(enteredTextArray[3]);
            }
            else if (consoleInput.contains("add standard"))
            {
                String[] enteredTextArray = consoleInput.split(" ");
                Ticket ticket = new Ticket();
                ticket.CustomerType = Ticket.CustomerTypes.Standard;
                ticket.Name = enteredTextArray[2];
                ticket.Age = Integer.parseInt(enteredTextArray[3]);
                Tickets.Add(ticket);
            }
            else if (consoleInput.contains("add student"))
            {
                String[] enteredTextArray = consoleInput.split(" ");
                Ticket ticket = new Ticket();
                ticket.CustomerType = Ticket.CustomerTypes.Student;
                ticket.Name = enteredTextArray[2];
                ticket.Age = Integer.parseInt(enteredTextArray[3]);
                Tickets.Add(ticket);
            }
            else if (consoleInput.startsWith("exit"))
                continue;
            else
                out.println("Unknown Input: " + consoleInput);
        } while (!consoleInput.startsWith("exit"));
    }
}

File: nodejs/ticketbooker/test/consoleapp.js
'use strict';

var assert = require('assert');
var ConsoleApp = require('../lib/consoleapp');
var newLine = require('os').EOL;

describe('ticketbooker regression tests', function () {
  it('input doggy value, run console, unknown input returned.', function () {
    var actual = runConsoleApp('gobbledygook');
    var expected = 'Unknown Input: gobbledygook';
    assert.equal(actual, expected);
  });

  it('input ticket price and tickets and print summary, run console, summary returned', function () {
    var input = ['set ticket price 10',
      'add standard Fred 30',
      'add standard Sarah 54',
      'add student Bob 19',
      'add student Kate 17',
      'add student Ashleigh 20',
      'print summary'];
    var actual = runConsoleApp(input);
    var expected = '** SALES REPORT **' + newLine + newLine;
    expected += 'Number of Tickets Sold' + newLine;
    expected += '  Standard: 2' + newLine;
    expected += '  Student: 3' + newLine + newLine;
    expected += 'Income' + newLine;
    expected += '  Standard Total = £20' + newLine;
    expected += '  Student Total = £18' + newLine;
    expected += '  Total = £38';
    assert.equal(actual, expected);
  });

  function runConsoleApp(consoleInput) {
    var stdMocks = require('std-mocks');
    var stdin = require('mock-stdin').stdin();
    var consoleApp = new ConsoleApp();
    consoleApp.start();
    stdin.send(consoleInput);
    stdMocks.use();
    stdin.send(newLine);
    stdMocks.restore();
    stdin.send('exit' + newLine);
    stdin.restore();
    var output = stdMocks.flush();
    var outputClean = output.stdout.toString().replace(',', newLine).trim();
    return outputClean;
  }
});
